src/routes/auth.route.ts

```diff
@@ -0,0 +1,117 @@
+import express, { type Router } from 'express'
+import * as AuthController from '../controllers/auth.controller'
+
+const router: Router = express.Router()
+
+/**
+ * @swagger
+ * tags:
+ *   - name: authentication
+ *     description: User authentication
+ */
+
+/**
+ * @swagger
+ * definitions:
+ *   Login:
+ *     type: object
+ *     properties:
+ *       email:
+ *         type: string
+ *         example: test@gmail.com
+ *       password:
+ *         type: string
+ *         example: "1234"
+ *   Token:
+ *    type: object
+ *    properties:
+ *      email:
+ *        type: string
+ *        example: test@gmail.com
+ *      token:
+ *        type: string
+ *        example: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJhZG1pbkBhZG1pbi5jb20iLCJpYXQiOjE1MDk5ODg2NDZ9.1zTKAzXmuyQDHw4uJXa324fFS1yZwlriFSppvK6nOQY
+ *   Error:
+ *      type: object
+ *      properties:
+ *         message:
+ *            type: string
+ *         error:
+ *            type: boolean
+ *            default: true
+ *
+ */
+
+/**
+ * @swagger
+ * /auth/sign-up:
+ *   post:
+ *     tags: ['User']
+ *     summary: Create a new user
+ *     description: Create a new user
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               name:
+ *                 type: string
+ *                 description: User's name
+ *                 example: Toto
+ *               email:
+ *                 type: string
+ *                 description: User's email
+ *                 example: Toto@gmail.com
+ *               password:
+ *                 type: string
+ *                 description: User's password
+ *                 example: azerty
+ *     responses:
+ *       201:
+ *         description: The created user
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/User'
```

Comment:
i don't know if it will work, since `#/components/schemas/User` is declared in another file. can you check on the api-docs interface, if it's working ?